consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/RoundChangeMessageValidator.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.ibft.validation;
+
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.IbftPreparedCertificate;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.IbftSignedMessageData;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.IbftUnsignedPrepareMessageData;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.IbftUnsignedRoundChangeMessageData;
+import tech.pegasys.pantheon.ethereum.core.Address;
+
+import java.util.Collection;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class RoundChangeMessageValidator {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final MessageValidator messageValidator;
+  private final Collection<Address> validators;
+
+  public RoundChangeMessageValidator(
+      final MessageValidator messageValidator, final Collection<Address> validators) {
+    this.messageValidator = messageValidator;
+    this.validators = validators;
+  }
+
+  public boolean validateMessage(
+      final IbftSignedMessageData<IbftUnsignedRoundChangeMessageData> msg) {
+
+    if (!validators.contains(msg.getSender())) {
+      LOG.info(
+          "Invalid RoundChange message, was not transmitted by a validator for the "
+              + "associated round.");
+      return false;
+    }
+
+    if (!msg.getUnsignedMessageData().getPreparedCertificate().isPresent()) {
+      LOG.info("RoundChange message is valid. It contains no PrepareCertificate.");
+      return true;
+    }
+
+    final IbftPreparedCertificate certificate =
+        msg.getUnsignedMessageData().getPreparedCertificate().get();
+
+    if (!messageValidator.addPreprepareMessage(certificate.getIbftPrePrepareMessage())) {
+      LOG.info("Invalid Round Change message, embedded Preprepare message failed validation.");
```

Comment:
Remove space in RoundChange like other messages

Reply:
Done.